File: tests/integration/test_ui_options.py
# SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
# SPDX-License-Identifier: Apache-2.0
"""Tests for different UI modes."""

import pytest

from tests.integration.conftest import AIPerfCLI
from tests.integration.models import FakeAIServer


@pytest.mark.integration
@pytest.mark.asyncio
class TestUIOptions:
    """Tests for different UI modes."""

    async def test_none_ui(self, cli: AIPerfCLI, fakeai_server: FakeAIServer):
        """None UI mode (no interactive output)."""
        result = await cli.run(
            f"""
            aiperf profile \
                --model deepseek-ai/DeepSeek-R1 \
                --url {fakeai_server.url} \
                --endpoint-type chat \
                --request-count 10 \
                --concurrency 2 \
                --workers-max 1 \
                --ui none
            """
        )
        assert result.request_count == 10
        assert result.has_all_outputs

File: rust/integration/tests/test_ui_options.rs
// SPDX-FileCopyrightText: Copyright (c) 2026 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
// SPDX-License-Identifier: Apache-2.0
#[path = "common/mod.rs"]
mod common;
use common::*;

/// None UI mode (no interactive output).
#[tokio::test]
async fn test_none_ui() {
    let h = AIPerfHarness::new().await;
    let r = h.run(&format!(
        "--model {DEFAULT_MODEL} --url {} --endpoint-type chat \
         --request-count 10 --concurrency 2 --workers-max 1 --ui none",
        h.mock.url
    ));
    assert!(r.success());
    assert_eq!(r.artifacts.request_count() as u32, 10);
}
